lib/puppet/type/file/ensure.rb

```diff
@@ -175,6 +183,14 @@ def retrieve
     end
 
     def sync
+      if self.should == :absent
+        # Make a dummy content property so that we can
```

Comment:
> What's the content property that exists here otherwise, and how is it set?

Before we do this, it's `nil`.  When you put `:absent` into `@resource[:content]`, Puppet generates the correct object to go in there.

> It seems like you have to set this so Puppet will read the checksum of the file.

Pretty much.